### ISSUE TITLE
AFK players should be replaced by AI

### ISSUE THREAD

**Issue by Kurtzmusch** [solaris-games/solaris#528]
- might be useful for beginner players to learn the game on a quick match against AI as well

rough outline of AI behavior: 
- should attempt to capture unclaimed stars
- should build industry and science as soon as the cycle begins
- should delay building economy until cycle is near end
- buying strategy 3:1:1 early game -> 3:2:1 early mid game -> 3:3:2 mid game -> 5:4:2 late game (E, I, S)
- if buying strategy gets to hard to figure out, it can try to copy whatever the best players are doing on the match
- should move ships into border
- should trade back tech if not enemy
- should consider a player an enemy if it gets attacked , unless the attacker launched the attack before the AI owned the star AND had no vision of the AI carrier that conquered the star
- should actively seek trade from low star count players(AIs included)
- should attempt to defend itself if it can do so before combat takes place

this list covers a very basic and passive AI. after these gets implemented, we can think about refining the behavior and making it more polished and reactive.

**Comment by mike-eason** [comment]
This issue is so overwhelmingly complex that we need to split this up into smaller tasks. I'll set up a project board for it and we can discuss the first steps.

**Comment by Kurtzmusch** [comment]
i think the first step is to add a foundation for AI to be able to play, basically somewhere on the game tick calls an AIservice, or an array of AIs to take action/play. most ticks the AI wont do anything. we can have an issue for each major component on that list, and try to work in parallel. there could be a main play() function which then calls the individual decision functions, for example:
```
play(game, ownedStars, ownedCarriers) {
 respondToTrade()
 seekClaimableStars()
 defendFromAttacks()
 setupCarrierLoops()
 //now with the remaining money
 buildInfrastructure()
}
```
having instances AIs instead of an AIservice that play for all afk empires might be interesting since it can retain state, for example: which stars are border, which players are allyes/enemyes, and so on. AIs will also like to retain state since its common for AIs to be a 'state machine'. this means an AI that is under attack might build infrastructure differently then one that is not.

since the player API and the services that manipulate the main game data structure are separate, the AI should probably interact directly with those services.

its important that the AI plays before generating the reports, so we dont need to resend stuff after the AI plays

**Comment by mike-eason** [comment]
**Stage One**

- A player is considered to be AI if they have been defeated or are AFK.
- All existing carrier waypoints and loops will not be removed if a player turns into AI.
- All existing star warp gates will not be destroyed if a player turns into AI.
- On the first tick after production, spend x% (50%?) of funds on ind/sci reserve the rest for trades and eco.
- On the last tick of a production cycle, spend all remaining funds on economy.
- All positive and negative interactions will be kept track of, even if the player isn't AI - So that if the player does become AI, then that data set is already prepared.
- If a player has a positive standing with an AI and trades tech/cash, then there will be a random chance that the AI will trade back. This will be calculated as 50% + (standing * 5%) where standing is a value from -10 to +10.
- A player can increase their standings with an AI by either sending technology or a given amount of cash (formula to be finalised).
- The AI will pick research at random. - It would be desirable if the AI could figure out which tech was the least of all players and research that so it keeps up.
